R/utils.R

```diff
@@ -14,3 +14,23 @@ list_repo <- function(orgname = 'natverse') {
   repos <- gh::gh(paste0("/orgs/",orgname,"/repos"), type = "public")
   vapply(repos, "[[", "", "name")
 }
+
+have_xquartz <- function() {
+  isTRUE(nzchar(Sys.which('Xquartz')))
+}
+
+system_requirements_ok <- function() {
+  if(isTRUE(.Platform$OS.type == "windows")) {
+    return(TRUE)
+  }
+  sysname=Sys.info()["sysname"]
+  if(isTRUE(sysname=='Darwin')) {
+    if(!have_xquartz()){
+      usethis::ui_todo(paste("Please download and install Xquartz from",
+                       "https://www.xquartz.org!\nThis is a system requirement",
+                       "and is needed for 3D display of neurons."))
+      return(FALSE)
+    }
+  }
+  return(TRUE)
+}
```

Comment:
```suggestion
system_requirements_ok <- function() {
  if(grepl("linux-gnu", R.version$os) | grepl("mingw", R.version$os)) {
    return(TRUE)
  }
  if(grepl("^darwin", R.version$os)) {
    if(!have_xquartz()){
      usethis::ui_todo(paste("Please download and install Xquartz from",
                       "https://www.xquartz.org!\nThis is a system requirement",
                       "and is needed for 3D display of neurons."))
      return(FALSE)
    }
    return(TRUE)
  }
}
```

Reply:
Just a suggestion to streamline the way we detect OS and added support for linux in case it is needed in future for some other stuff.

Reply:
Thanks. Sorry I missed this at the time. I went with `Sys.info()["sysname"]` because Yihui Xie used it here

https://github.com/yihui/xfun/blob/db7f31709007905c8450f50ee532c36863e02fb5/R/os.R#L21

but I confess I don't know if there is any advantage over `grepl("^darwin", R.version$os)`